backend/data_preparation/crawler/twitter_filter_api_crawler.py

```diff
@@ -60,18 +61,21 @@ def crawl(self, keywords: List, batch_number: int = 100) -> List[int]:
                     self.reset_wait_time()
 
                     has_keywords = set(self.keywords) & self._tokenize_tweet_text(tweet)
-                    if tweet.get('retweeted_status') and set(self.keywords) & self._tokenize_tweet_text(
-                            tweet['retweeted_status']):
+                    if tweet.get('retweeted_status') and set(keyword.lower() for keyword in self.keywords) \
+                            & self._tokenize_tweet_text(tweet['retweeted_status']):
                         self._add_to_batch(tweet['retweeted_status']['id'])
 
+                    # if the tweet is sent within US and contains keywords, add its id to cache and data (for return)
                     elif tweet.get('place') and tweet['place']['country_code'] == "US" and has_keywords:
                         self._add_to_batch(tweet['id'])
                     else:
                         continue
 
-                    count = len(self.data)
-                    if count % (batch_number // 10) == 0:
-                        logger.info(f"Crawled ID count in this batch: {count}")
+                    # print Crawling info once every 10 tweets are collected
```

Comment:
10 is depending on the batch number. not a static number.
please update the comment